### DISCUSSION TITLE
Raw File URL Of Private Repository

### DISCUSSION THREAD

**Discussion by theabbie** [community/community#23845] [New to GitHub]
<p>When I click on the Raw file Button on a file on my private Repo, The Download link is of the form</p>
<p><code>https://raw.githubusercontent.com/&lt;user&gt;/&lt;repo&gt;/master/&lt;file&gt;?token=&lt;token&gt;</code></p>
<p>This link works also when I am not logged in, There is a token attached to the URL.</p>
<p>I wanted to know if that token is supposed to be kept secret or is it safe to share the link in case I want to share a file from private Repository?</p>

**Comment by nethgato** [comment, marked as answer]
<p>Hi <a class="mention" href="/theabbie">@theabbie</a> o/</p>
<p>I went ahead and re-categorized this post, but if you feel there is an API action that you’re trying to take with this raw URL, then please let me know and we can switch back :bow:</p>
<p>Though for your question:</p>
<blockquote>
<p>I wanted to know if that token is supposed to be kept secret or is it safe to share the link in case I want to share a file from private Repository?</p>
</blockquote>
<p>Yes! There’s really no reason why this would be unsafe to share, AFAIK. When we look at this post:</p>
<aside class="quote" data-post="2" data-topic="2958">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/token-expiration-on-urls-to-private-repos/2958/2">Token Expiration on URLs to Private Repos</a> <a class="badge-wrapper  bullet" href="/c/github-help/how-to-use-git-and-github/21"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #ffd1ac;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Ask and answer questions regarding Git and GitHub.">How to use Git and GitHub</span></a>
  </div>
  <blockquote>
    Hi <a class="mention" href="/shaigluskin">@shaigluskin</a>, 
Thank you for being here! The token will expire in 7 days.
  </blockquote>
</aside>

<p>…we can see:</p>
<blockquote>
<p>The token will expire in 7 days.</p>
</blockquote>
<p>So you won’t be able to keep that URL permanently, as it will change.</p>
<p>Please let us know if this is helpful and if you have any remaining questions or concerns!</p>

**Comment by theabbie** [comment]
<p>Awesome, Thank you for clarifying my doubt</p>

**Comment by base698** [comment]
<p>How is that token generated?  I’m trying to access my gist via the Github API.</p>

**Comment by airtower-luna** [comment]
<p>For a gist you shouldn’t need a token, those are available as long as you have the URL. Unless you access so many files that you might get in trouble with rate limiting. :wink:</p>
<p>In that case, or to download files from a private repository you’ll need a PAT, see <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token">Creating a personal access token</a>. Note that the PAT is not added to the URL, you need to add an <code>Authorization</code> header to the HTTP request as discussed here:</p>
<aside class="quote" data-post="1" data-topic="148553">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/how-can-a-github-workflow-read-a-file-on-another-private-repository-hosted-on-the-same-github-account/148553">How can a Github workflow read a file on another private repository hosted on the same GitHub account?</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    I have a private git repository like: 
Github.com/acct/repo1 
I have a test.json file in another private repository in the same GitHub subscription and account like: 
Github.com/acct/repo2/test.json 
How can a workflow in repo1 have http get access to the url Github.com/acct/repo2/test.json ? 
All I need is a simple way to read the content of test.json in a workflow running on repo2 . 
It works the best for me to read the file using a curl get request.
  </blockquote>
</aside>

**Comment by sankouski-dzmitry** [comment]
<p>Can I somehow get permament url to the file in private repository?</p>
<p>I’m using jenkins with configuration-as-code plugin, and I want it to look at github for configuration file.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="sankouski-dzmitry" data-post="6" data-topic="130512">
<div class="title">
<div class="quote-controls"></div>
 sankouski-dzmitry:</div>
<blockquote>
<p>Can I somehow get permament url to the file in private repository?</p>
</blockquote>
</aside>
<p>Not using <em>only</em> a URL. The file URL (without the <code>token=...</code> part) is permanent though, so if you provide the Jenkins job with a PAT and send an <code>Authorization</code> header it will work. :slightly_smiling_face:</p>

**Comment by twentyTwo** [comment]
Do you know how often the token changes?
